glslang/MachineIndependent/preprocessor/PpContext.h

```diff
@@ -456,16 +471,32 @@ class TPpContext {
         {
             prevScanner = pp->parseContext.getScanner();
             pp->parseContext.setScanner(&scanner);
+            pp->push_include(includedFile_);
+        }
+
+        void notifyDeleted() override
+        {
+          pp->parseContext.setScanner(prevScanner);
+          pp->pop_include();
         }
-        void notifyDeleted() override { pp->parseContext.setScanner(prevScanner); }
 
     private:
-        // Stores the titular string.
-        const std::string str_;
-        // Will point to str_[0] and be passed to scanner constructor.
-        const char* const strings;
+        // Stores the prologue for this string.
+        const std::string prologue_;
+
+        // Stores the epilogue for this string.
+        const std::string epilogue_;
+
+        // Points to the IncludeResult that this TokenizableString represents.
+        TShader::Includer::IncludeResult* includedFile_;
+
+        // Will point to prologue_, includedFile_->file_data and epilogue_
```

Comment:
Probably worth saying that these pointers (strings and names arrays) don't own the storage they reference.  Sure that's somewhat implicit, but it's best to say it explicitly here.